bin/src/Flags.purs

```diff
@@ -2,229 +2,304 @@ module Spago.Bin.Flags where
+censorBuildWarnings :: Parser (Maybe Core.CensorBuildWarnings)
 censorBuildWarnings =
-  ArgParser.argument [ "--censor-build-warnings" ]
-    "Censor compiler warnings based on file's location: 'dependency', 'project', or 'all'"
-    # ArgParser.unformat "ARG"
-        ( case _ of
+  O.optional $
+    flip
+      O.option
+      ( O.long "censor-build-warnings"
+          <> O.help "Censor compiler warnings based on file's location: 'dependency', 'project', or 'all'"
+          <> O.metavar "ARG"
+      )
+      ( O.eitherReader
+          case _ of
             "all" -> Right Core.CensorAllWarnings
             "project" -> Right Core.CensorProjectWarnings
             "dependency" -> Right Core.CensorDependencyWarnings
             _ -> Left $ "Expected 'all', 'project', or 'dependency'"
-        )
-    # ArgParser.optional
+      )
 
-showSource ∷ ArgParser (Maybe ShowSourceCode)
+showSource :: Parser (Maybe ShowSourceCode)
 showSource =
-  ArgParser.flag
-    [ "--no-source" ]
-    "Disable original source code printing"
-    $> Just NoSourceCode
-    # ArgParser.default Nothing
+  O.optional
+    $ O.switch
+        ( O.long "no-source"
+            <> O.help "Disable original source code printing"
+        )
+    $>
+      NoSourceCode
 
-censorCodes :: ArgParser (Maybe (NonEmptySet String))
+censorCodes :: Parser (Maybe (NonEmptySet String))
 censorCodes =
-  ArgParser.argument [ "--censor-code" ]
-    "Censor a specific error code (e.g. `ShadowedName`)"
-    # ArgParser.many
-    <#> NonEmptySet.fromFoldable
-
-filterCodes :: ArgParser (Maybe (NonEmptySet String))
+  NonEmptySet.fromFoldable
+    <$>
+      ( O.many
+          $ O.strOption
+              ( O.long "censor-code"
+                  <> O.metavar "CODE"
+                  <> O.help "Censor a specific error code (e.g. `ShadowedName`)"
+              )
+      )
+
+filterCodes :: Parser (Maybe (NonEmptySet String))
 filterCodes =
-  ArgParser.argument [ "--filter-code" ]
-    "Only show a specific error code (e.g. `TypesDoNotUnify`)"
-    # ArgParser.many
-    <#> NonEmptySet.fromFoldable
-
-statVerbosity :: ArgParser (Maybe Core.StatVerbosity)
-statVerbosity = ArgParser.optional $ ArgParser.choose "StatVerbosity"
-  [ Core.VerboseStats <$ ArgParser.flag [ "--verbose-stats" ] "Show counts for each warning type"
-  , Core.NoStats <$ ArgParser.flag [ "--censor-stats" ] "Censor warning/error summary"
-  ]
-
-persistWarnings ∷ ArgParser (Maybe Boolean)
+  NonEmptySet.fromFoldable
+    <$>
+      O.many
+        ( O.strOption
+            $ O.long "filter-code"
+            <> O.metavar "CODE"
+            <> O.help "Only show a specific error code (e.g. `TypesDoNotUnify`)"
+        )
+
+statVerbosity :: Parser (Maybe Core.StatVerbosity)
+statVerbosity =
+  O.optional
+    $ (Core.VerboseStats <$ O.switch (O.long "verbose-stats" <> O.help "Show counts for each warning type"))
+    <|>
+      (Core.NoStats <$ O.switch (O.long "censor-stats" <> O.help "Censor warning/error summary"))
```

Comment:
Some of the tests are showing a different output related to how the build stats are shown (and the `strict` flag as well), which makes me think that these might be behaving differently

Reply:
Yeah, using `$>` was inappropriate.